rslib/ascii_percent_encoding/src/lib.rs

```diff
@@ -82,7 +82,8 @@ impl AsciiSet {
     }
 
     fn should_percent_encode(&self, byte: u8) -> bool {
-        !byte.is_ascii() || self.contains(byte)
+        // MODIFIED
+        byte.is_ascii() && self.contains(byte)
```

Comment:
I'm afraid this doesn't look correct. If a unicode char is converted to utf8, some of the individual bytes will be below 128. Eg the last one in this kanji:

u"今".encode("utf8")
b'\xe4\xbb\x8a'

Instead of checking the bytes of the utf8, we need to operate on the unicode chars, and focus on only chars which are <128. Effectively string.chars().map(|c| if c.is_ascii { percent_escape(...) }).collect::<String>()

A unit test that checks a few test cases might be a good idea here :-)

Reply:
Yeah, UTF-8 is great. 🙂